packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx

```diff
@@ -1,40 +1,80 @@
-import {Component, h, Host, Listen, State} from '@stencil/core';
+import {Component, h, Host, State, Element, Method, Prop} from '@stencil/core';
 
-export interface UpdateLiveRegionEventArgs {
-  region: string;
-  message: string;
+export interface FindAriaLiveEventArgs {
+  element?: HTMLAtomicAriaLiveElement;
 }
 
 /**
  * The `atomic-aria-live` component notifies screen readers of changes in the search interface.
- * @internal
+ *
+ * We do not recommend dynamically adding/removing this component from the page. It is better to have this component during the initial render of the page.
  */
 @Component({
   tag: 'atomic-aria-live',
   shadow: false,
 })
 export class AtomicAriaLive {
-  @State() public regions: Record<string, string> = {};
+  @Element() private host!: HTMLAtomicAriaLiveElement;
+  @State() private message = '';
 
-  @Listen('atomic/accessibility/updateLiveRegion', {target: 'document'})
-  protected updateMessage({
-    detail: {message, region},
-  }: CustomEvent<UpdateLiveRegionEventArgs>) {
-    this.regions = {...this.regions, [region]: message};
+  /**
+   * Only the `atomic-aria-live` element with the greatest priority will be used to announce changes in the search interface.
+   *
+   * @internal
+   */
+  @Prop({reflect: true}) priority = 1;
+
+  private lastUpdatedRegion?: string;
+  private disconnectFindAriaLiveEvent?: () => void;
+
+  protected onFindAriaLive(args: FindAriaLiveEventArgs) {
+    if (args.element && args.element.priority > this.priority) {
+      return;
+    }
+    args.element = this.host;
+  }
+
+  /**
+   * @internal
+   */
+  @Method()
+  public async updateMessage(region: string, message: string) {
+    const wouldOverwriteAnotherRegion = region !== this.lastUpdatedRegion;
+    if (wouldOverwriteAnotherRegion) {
+      this.lastUpdatedRegion = region;
+      if (!message) {
```

Comment:
Shouldn't that be at the top of the function, as an early check and return ?

Is there a scenario where we would want to update the message to "nothing" ? That's the same as not changing it at all, I think ?

Reply:
Changing the message to nothing may cancel the message on some screen readers. The purpose of returning here is to allow a region to cancel its own message, but not another region's.